python/lsst/sims/catUtils/mixins/VariabilityMixin.py

```diff
@@ -238,32 +252,75 @@ def applyAgn(self, params, expmjd_in):
         sfint['z'] = params['agn_sfz']
         sfint['y'] = params['agn_sfy']
         tau = params['agn_tau']
-        epochs = expmjd - toff
-        if epochs.min() < 0:
+
+        # A string made up of this AGNs variability parameters that ought
+        # to uniquely identify it.
+        #
+        cache_name = '%d_%.12f_%.12f_%.12f_%.12f_%.12f_%.12f_%.12f_%.12f' \
+        %(seed, sfint['u'], sfint['g'], sfint['r'], sfint['i'], sfint['z'],
+          sfint['y'], tau, toff)
+
+        resumption = False
+
+        # Check to see if this AGN has already been simulated.
+        # If it has, see if the previously simulated MJD is
+        # earlier than the first requested MJD.  If so,
+        # use that previous simulation as the starting point.
+        #
+        if cache_name in _agn_lc_cache:
+            if _agn_lc_cache[cache_name]['mjd'] <expmjd:
+                resumption = True
+
+        if resumption:
+            rng = copy.deepcopy(_agn_lc_cache[cache_name]['rng'])
+            start_date = _agn_lc_cache[cache_name]['mjd']
+            dx_0 = _agn_lc_cache[cache_name]['dx']
+        else:
+            start_date = toff
+            rng = numpy.random.RandomState(seed)
+            dx_0 = {}
+            for k in sfint:
+                dx_0[k]=0.0
+
+        endepoch = expmjd - start_date
+
+        if endepoch < 0:
```

Comment:
Quirky observation: you do need the '+' signs once you start concatenating strings of the form

"text and %d" % my_variable

Given that this mixin will only be used to add variability to an AGN, it seems appropriate to me that the code should crash if variability is impossible (if the user wanted a static AGN, they could have made an InstanceCatalog that did not inherit from VariabilityGalaxies).  I am going to leave it for now.  We can discuss more later, if we want (there are a bunch of exceptions that are poorly handled in the Variability code).  It is a design decision that was made a while ago, and I am erring on the side of not doing anything.